lib/handlerTemplate.js

```diff
@@ -264,11 +268,24 @@ function prepareRequestChain(conf) {
 function createHandler(spec) {
     validateSpec(spec);
     var requestChain = prepareRequestChain(spec);
-    return function(restbase, req) {
-        return generatePromiseChain(restbase, {
-            request: req
-        }, requestChain);
-    };
+    if (requestChain.length === 1
+            && requestChain[0].length === 1
+            && requestChain[0][0].request
+            && !requestChain[0][0].return
+            && !requestChain[0][0].catch) {
```

Comment:
Could be nice to wrap this ugly condition in a function with a descriptive name.